msal/src/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -58,8 +58,16 @@ internal AuthenticationResult(MsalAccessTokenCacheItem msalAccessTokenCacheItem,
             _msalIdTokenCacheItem = msalIdTokenCacheItem;
             if (_msalAccessTokenCacheItem.HomeAccountId != null)
             {
-                Account = new Account(AccountId.FromClientInfo(_msalAccessTokenCacheItem.ClientInfo),
-                    _msalIdTokenCacheItem?.IdToken?.PreferredUsername, _msalAccessTokenCacheItem.Environment);
+                if (_msalAccessTokenCacheItem.TenantId == null)
+                {
+                    //Null tenant is indicates Adfs authority
+                    Account = new Account(new AccountId(_msalAccessTokenCacheItem.HomeAccountId), _msalIdTokenCacheItem?.IdToken.Upn, _msalAccessTokenCacheItem.Environment);
```

Comment:
>IdToken [](start = 121, length = 7)

THis means you are sure the IdToken cannot be null. If so, please leave as is, we'll get a NullRef exception which is easy to track